### DISCUSSION TITLE
Need help choosing a name for a new particle node

### DISCUSSION THREAD

**Discussion by AlexandreMinotCSI** [CGNS/CGNS#764] [CGNStalk]
Dear CGNS community,

We are working on a new proposal for CPEX0046 to unable storing of particle data in CGNS. In particular, we are adding a node under the base to hold groups of particles. This node would contain, among others, _ParticleCoordinates_t_ and _ParticleSolutions_t_ to specify the location of each particle in the group and related data. Multiple groups of particle can be created under the base, to discern for example blue and green particles.

We are having trouble coming up with a good name for the top node that will hold each group/set of particles. Our top candidates are:
- _ParticleGroup_t_
- _ParticleZone_t_
- _ParticleCloud_t_
- _DisctretePhase_t_

What are your thoughts one this?

Many thanks,
Alexandre

**Comment by MicK7** [comment]
My ordered preference :

- ParticleCloud_t looks nice. It seems generic as you can imagine having multiple ParticleCloud_t nodes (sharing physical or numerical properties through a Family_t node) . It is different from existing names and do not create confusion (can help parsers too).

- ParticleZone_t looks similar to existing Zone_t node. Meaning a zone that contains particles.

- DiscretePhase_t .  Very targeted naming, explicit while not very generic. So why not ! 

- ParticleGroup_t : not in favor at all, "grouping" is why Family_t nodes were introduced and extended in CGNS. Thus this naming lead to confusion.


Are you also having nodes for "Interpolation scheme" attached to the particle simulation ?

**Reply by brtnfld** [reply]
I also like (1) and (2), maybe (2) more since it builds on an already conventional one.

**Comment by AlexandreMinotCSI** [comment]
The only thing I don't like about ParticleCloud_t is that a cloud has a physical meaning: liquid particles in suspension in a gas. I would not want people to infer that meaning. But maybe that is not a real issue? What do you think?

**Reply by MicK7** [reply]
You argument is sound.
In this case, what about a "DiscreteZone_t" node, since particles are a discrete set of data ?  This would be very generic...

**Reply by AlexandreMinotCSI** [reply]
Yes, that it also an idea. The thing is that the normal Zone_t is not continuous, since the mesh is discrete. Also, it makes the existing    DiscreteData_t node even more confusing. We would have:
- Base_t/Zone_t/DiscreteData_t
- Base_t/DiscreteZone_t/ParticleData_t

What do you think?

It's one of the issue I have: Particles are not really more discrete than a mesh when it comes to the CFD data. The real difference is that they are an unorganized sets of points, as opposed to a set of connected elements. Physically, particles are discrete compared to the continuous media that holds them. So discrete phase, as opposed to continuous phase, could be acceptable. But maybe staying clear of the word discrete is better?

**Reply by AlexandreMinotCSI** [reply]
ParticleCloud_t is a good candidate, because it is very clear to a reader what the data is. This is pretty clear:

- Base_t/Zone_t/FlowSolution_t
- Base_t/ParticleCloud_t/ParticleSolution_t

**Reply by AlexandreMinotCSI** [reply]
This also looks good:

- Base_t/Zone_t/FlowSolution_t
- Base_t/ParticleZone_t/ParticleSolution_t

**Reply by smguzik** [reply]
My preference is for ParticleZone_t

**Comment by AlexandreMinotCSI** [comment]
Looks like _ParticleZone_t_ is the name with the most consensus and least amount of objections, so let's go with that.

@MicK7 , What do you mean by Interpolation Scheme? Do you mean what model is being used to resolve particles? Yes we intend to allow that. This would in particular cover one of the needs of the original CPEX. We have only just started on the subject so I don't have much details. Look like simply allowing FlowEquationSet_t and GoverningEquations_t as children of ParticleZone_t will go a long way. We will ParticleModel_t as a subset of ModelType_t to hold model names. We plan on putting as much information as we can from our CFD code and see what nodes we need to hold that information.
